internal/patroni/api.go

```diff
@@ -177,3 +178,24 @@ func (exec Executor) RestartPendingMembers(ctx context.Context, role, scope stri
 
 	return err
 }
+
+// GetTimeline gets the patronictl status and returns the timeline,
+// currently the only information required by the postgres-operator.
+func (exec Executor) GetTimeline(ctx context.Context) (string, error) {
+	var stdout, stderr bytes.Buffer
+
+	// The following exits zero when it is able to read the DCS and communicate
+	// with the Patroni HTTP API. It prints the result of calling "GET /cluster"
+	// - https://github.com/zalando/patroni/blob/v2.1.1/patroni/ctl.py#L849
+	err := exec(ctx, nil, &stdout, &stderr,
+		"patronictl", "list")
+
+	regEx := regexp.MustCompile(`\|\s+Leader\s+\|\s+running\s+\|\s+(\d+)\s+\|`)
+	timeline := regEx.FindStringSubmatch(stdout.String())
```

Comment:
Couple notes here:

(a) I'm swallowing stderr here because I've not seen anything interesting, but maybe if it has something we should exit like an err?
(b) If we have any use for `patronictl list`, then we might just want to pass stdout through and leave the parsing for the calling func?